File: iosApp/iosApp/Extensions/AuthErrorExtensions.swift
//
//  AuthErrorExtensions.swift
//  iosApp
//
//  Extension to convert Kotlin AuthError to localized error messages
//
import Shared

extension Shared.AuthError {
    func toLocalizedMessage() -> String {
        switch self {
        case is Shared.AuthError.InvalidCredentials:
            return NSLocalizedString("error_invalid_credentials", comment: "")
        case is Shared.AuthError.EmailNotConfirmed:
            return NSLocalizedString("error_email_not_confirmed", comment: "")
        case is Shared.AuthError.NoConnection:
            return NSLocalizedString("error_no_connection", comment: "")
        case is Shared.AuthError.RateLimitExceeded:
            return NSLocalizedString("error_rate_limit_exceeded", comment: "")
        case is Shared.AuthError.UserNotFound:
            return NSLocalizedString("error_user_not_found", comment: "")
        case is Shared.AuthError.WeakPassword:
            return NSLocalizedString("error_weak_password", comment: "")
        case is Shared.AuthError.UserAlreadyExists:
            return NSLocalizedString("error_user_already_exists", comment: "")
        case is Shared.AuthError.AuthenticationFailed:
            return NSLocalizedString("error_authentication_failed", comment: "")
        default:
            return NSLocalizedString("error_unexpected", comment: "")
        }
    }
}

File: composeApp/src/androidMain/kotlin/com/ivangarzab/kluvs/ui/extensions/AuthErrorExtensions.kt
package com.ivangarzab.kluvs.ui.extensions

import androidx.compose.runtime.Composable
import androidx.compose.ui.res.stringResource
import com.ivangarzab.kluvs.R
import com.ivangarzab.kluvs.auth.domain.AuthError

/**
 * Maps [com.ivangarzab.kluvs.auth.AuthError] codes to localized string resources.
 *
 * Usage in Composables:
 * ```
 * val errorMessage = error.toLocalizedMessage()
 * Text(text = errorMessage)
 * ```
 */
@Composable
fun AuthError.toLocalizedMessage(): String = when (this) {
    is AuthError.InvalidCredentials -> stringResource(R.string.error_invalid_credentials)
    is AuthError.EmailNotConfirmed -> stringResource(R.string.error_email_not_confirmed)
    is AuthError.NoConnection -> stringResource(R.string.error_no_connection)
    is AuthError.RateLimitExceeded -> stringResource(R.string.error_rate_limit_exceeded)
    is AuthError.UserNotFound -> stringResource(R.string.error_user_not_found)
    is AuthError.WeakPassword -> stringResource(R.string.error_weak_password)
    is AuthError.UserAlreadyExists -> stringResource(R.string.error_user_already_exists)
    is AuthError.AuthenticationFailed -> stringResource(R.string.error_authentication_failed)
    is AuthError.UnexpectedError -> stringResource(R.string.error_unexpected)
}
